File: rest/src/main/java/my/sample/rest/impl/SampleInvoker.java
package my.sample.rest.impl;

import my.sample.activator.SampleLogger;

/**
 * @author mohammad shamsi <m.h.shams@gmail.com>
 */
public class SampleInvoker { //extends JAXRSInvoker {
    private SampleLogger logger;

    @SuppressWarnings("UnusedDeclaration")
    public void setLogger(SampleLogger logger) {
        this.logger = logger;
    }

//    @Override
//    public Object invoke(Exchange exchange, Object request, Object resourceObject) {
//        logger.warn("Invoker...");
//        return super.invoke(exchange, request, resourceObject);
//    }
}

File: rest/src/main/kotlin/my/sample/rest/impl/SampleInvoker.kt
package my.sample.rest.impl

import my.sample.activator.SampleLogger
import org.apache.cxf.jaxrs.JAXRSInvoker
import org.apache.cxf.message.Exchange

/**
 * @author mohammad shamsi <m.h.shams@gmail.com>
 */
public class SampleInvoker : JAXRSInvoker() {
    public var logger: SampleLogger? = null

    override fun invoke(exchange: Exchange?, request: Any?, resourceObject: Any?): Any? {
        logger?.warn("Invoker...")
        return super.invoke(exchange, request, resourceObject)
    }
}
